core/src/datayoga_core/step_buffer.py

```diff
@@ -0,0 +1,55 @@
+import asyncio
+import datetime
+import logging
+
+from datayoga_core.step import Step
+
+logger = logging.getLogger("dy")
+
+class StepBuffer(Step):
+    def __init__(self, id, min_buffer_size=4, max_buffer_size=4, flush_ms=None):
+        super().__init__(id, None)
```

Comment:
Change in Step so the Block will be optional:
`def __init__(self, id: str, block: Optional[Block], concurrency=1):`